airsonic-main/src/main/java/org/airsonic/player/service/TranscodingService.java

```diff
@@ -195,27 +193,46 @@ public Parameters getParameters(MediaFile mediaFile, Player player, Integer maxB
 
         Parameters parameters = new Parameters(mediaFile, videoTranscodingSettings);
 
-        TranscodeScheme transcodeScheme = getTranscodeScheme(player);
-        if (maxBitRate == null && transcodeScheme != TranscodeScheme.OFF) {
-            maxBitRate = transcodeScheme.getMaxBitRate();
+        if (maxBitRate == null) {
+            maxBitRate = TranscodeScheme.OFF.getMaxBitRate();
         }
 
+        TranscodeScheme transcodeScheme = getTranscodeScheme(player).strictest(TranscodeScheme.fromMaxBitRate(maxBitRate));
+        maxBitRate = transcodeScheme.getMaxBitRate();
+
         boolean hls = videoTranscodingSettings != null && videoTranscodingSettings.isHls();
         Transcoding transcoding = getTranscoding(mediaFile, player, preferredTargetFormat, hls);
-        if (transcoding != null) {
-            parameters.setTranscoding(transcoding);
-            if (maxBitRate == null) {
-                maxBitRate = mediaFile.isVideo() ? VideoPlayerController.DEFAULT_BIT_RATE : TranscodeScheme.MAX_192.getMaxBitRate();
+        Integer bitRate = mediaFile.getBitRate();
+        if (bitRate == null) {
+            // Assume unlimited bitrate
+            bitRate = TranscodeScheme.OFF.getMaxBitRate();
+        }
+
+        if (mediaFile.isVideo()) {
+            if (maxBitRate == 0) {
+                maxBitRate = VideoPlayerController.DEFAULT_BIT_RATE;
+            }
+        } else {
+            if (mediaFile.isVariableBitRate()) {
+                // Assume VBR needs approx 20% more bandwidth to maintain equivalent quality in CBR
+                bitRate = bitRate * 6 / 5;
```

Comment:
Fair enough, without a first pass it's impossible to determine in advance. Might need a note in the docs though.

Reply:
Yes, updating the docs would be good.  Happy to create a PR to do that once this is merged (finalized).